diff --git a/protobuf-matchers/protocol-buffer-matchers.cc b/protobuf-matchers/protocol-buffer-matchers.cc
index 4552084..0123456 100644
--- a/protobuf-matchers/protocol-buffer-matchers.cc
+++ b/protobuf-matchers/protocol-buffer-matchers.cc
@@ -49,14 +49,14 @@ class StringErrorCollector : public google::protobuf::io::ErrorCollector {
   explicit StringErrorCollector(std::string* error_text)
       : error_text_(error_text) {}
 
-  void AddError(int line, int column, const std::string& message) override {
-    absl::SubstituteAndAppend(error_text_, "$0($1): $2\n", line, column,
-                              message.c_str());
+  void RecordError(int line, google::protobuf::io::ColumnNumber column, absl::string_view message) override {
+    absl::SubstituteAndAppend(error_text_, "$0($1): $2\n", line, (int)column,
+                              (std::string)message);
   }
 
-  void AddWarning(int line, int column, const std::string& message) override {
-    absl::SubstituteAndAppend(error_text_, "$0($1): $2\n", line, column,
-                              message.c_str());
+  void RecordWarning(int line, google::protobuf::io::ColumnNumber column, absl::string_view message) override {
+    absl::SubstituteAndAppend(error_text_, "$0($1): $2\n", line, (int)column,
+                              (std::string)message);
   }
 
  private: