diff --git a/extensions/bindgen/private/bindgen.bzl b/extensions/bindgen/private/bindgen.bzl
--- private/bindgen.bzl
+++ private/bindgen.bzl
@@ -144,17 +144,12 @@ def _generate_cc_link_build_info(ctx, cc_lib):
     for linker_input in cc_lib[CcInfo].linking_context.linker_inputs.to_list():
         for lib in linker_input.libraries:
-            if lib.static_library:
-                rustc_flags.append("-lstatic={}".format(get_lib_name_default(lib.static_library)))
-                linker_search_paths.append(lib.static_library.dirname)
-                compile_data.append(lib.static_library)
-            elif lib.pic_static_library:
+            if lib.pic_static_library:
                 rustc_flags.append("-lstatic={}".format(get_lib_name_default(lib.pic_static_library)))
                 linker_search_paths.append(lib.pic_static_library.dirname)
                 compile_data.append(lib.pic_static_library)
-
-    if not compile_data:
-        fail("No static libraries found in {}".format(
-            cc_lib.label,
-        ))
+            elif lib.static_library:
+                rustc_flags.append("-lstatic={}".format(get_lib_name_default(lib.static_library)))
+                linker_search_paths.append(lib.static_library.dirname)
+                compile_data.append(lib.static_library)
 
@@ -233,6 +228,14 @@ def _rust_bindgen_impl(ctx):
 
     # Configure Clang Arguments
     args.add("--")
+    resource_dir = libclang_dir + "/clang/10.0.0"
+    for f in libclang[DefaultInfo].data_runfiles.files.to_list():
+        if f.basename == "stddef.h":
+            if f.dirname.endswith("/include"):
+                resource_dir = f.dirname[:-8]
+                break
+    args.add("-resource-dir", resource_dir)
+    args.add("-isystem", resource_dir + "/include")
 
     compile_variables = cc_common.create_compile_variables(
         cc_toolchain = cc_toolchain,
@@ -340,6 +344,7 @@ def _rust_bindgen_impl(ctx):
             transitive = [
                 cc_lib[CcInfo].compilation_context.headers,
                 _get_libs_for_static_executable(libclang),
+                libclang[DefaultInfo].data_runfiles.files,
             ] + ([
                 _get_libs_for_static_executable(libstdcxx),
             ] if libstdcxx else []),
